packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -188,6 +195,50 @@ export interface IFluidDataStoreContextEvents extends IEvent {
 	(event: "attaching" | "attached", listener: () => void);
 }
 
+/**
+ * Eventually we should remove the delta manger from being exposed to Datastore runtimes via the context. However to remove that exposure we need to add new
+ * features, and those features themselves need forward and back compat. This proxy is here to enable that back compat. Each feature this proxy is used to
+ * support should be listed below, and as layer compat support goes away for those feature, we should also remove them from this proxy, with the eventual goal
+ * of completely remove this proxy.
+ *
+ * - Everything regarding readonly is to support older datastore runtimes which do not have the setReadonly function, so must get their readonly state via the delta manager.
+ *
+ */
+class ContextDeltaManagerProxy extends BaseDeltaManagerProxy {
+	constructor(base: IDeltaManagerFull) {
+		super(base, {
+			onReadonly: (readonly, reason): void => this.setReadonly(readonly, reason),
+		});
+		this._readonly = base.readOnlyInfo.readonly;
+	}
+
+	public get readOnlyInfo(): ReadOnlyInfo {
+		if (this._readonly === this.deltaManager.readOnlyInfo.readonly) {
+			return this.deltaManager.readOnlyInfo;
+		} else {
+			return this._readonly === true
+				? {
+						readonly: true,
+						forced: false,
+						permissions: undefined,
+						storageOnly: false,
+					}
+				: { readonly: this._readonly };
```

Comment:
nit (for symmetry with the other side of the ternary)
```suggestion
				: { readonly: false };
```